lib/bulk-apply-applications.js

```diff
@@ -243,91 +252,125 @@ function bulkApplyApplications (args, callback) {
+        locality: locality,
+        replyTo: dojo.email,
+        content: {
+          event: event,
+          dojo: dojo,
+          applicationDate: moment.utc(applications[0].created).format(eventDateFormat),
+          sessionName: session.name,
+          tickets: tickets,
+          status: applications[0].status,
+          applicationId: applications[iterator] ? applications[iterator].id : null,
+          eventDate: eventDate,
+          cancelLinkBase: protocol + '://' + zenHostname + '/dashboard/cancel_session_invitation'
+        }
+      };
+      async.parallel([
+        //Send parent and child emails
+        function sendApplicantsEmail () {
+          process.nextTick(function () {
+            emailSubject = emailSubjectData[applications[0].status];
+            async.eachSeries(profiles, function (profile, cb) {
+              //email payload changes for applicants
+              var payloadChanges = {
+                to: profile.email || null,
+                code: emailCode,
+                subject: emailSubject,
+                from: dojo.name+' <'+dojo.email+'>',
+                content: {
+                  applicantName: profile.name
+                }
+              };
+              //add common payload data
+              var payload = _.merge(payloadChanges, commonPayload);
+              iterator++;
+              if (!profile.email) return emailParents(cb);
 
-            function emailParents (cb) {
-              if (_.isEmpty(profile.parents)) return cb();
-              var parentsEmailed = [];
-              async.eachSeries(profile.parents, function (parent, cb) {
-                if (!_.isObject(parent)) {
-                  seneca.act({role: 'cd-users', cmd: 'load', id: parent, user: requestingUser}, function (err, parentUser) {
-                    if (err) return cb(err);
-                    payload.to = parentUser.email;
-                    if (!_.contains(parentsEmailed, payload.to) && payload.to !== profile.email) {
-                      parentsEmailed.push(payload.to);
-                      seneca.act({role: 'cd-dojos', cmd: 'send_email', payload: payload}, cb);
-                    } else {
-                      return cb();
-                    }
-                  });
-                } else {
+              //send the email
+              seneca.act({role: 'kue-queue', cmd: 'enqueue', name: 'bulk-apply-applications-kue', msg: _.clone({role: 'cd-dojos', cmd: 'send_email', payload: payload}, function (err, res) {
+                  if (err) return cb(err);
+                  if (applications[0].status === 'approved' || applications[0].status === 'pending' || applications[0].status === 'cancelled' ) {
+                    emailParents(cb);
+                  } else {
+                    return cb();
+                  }
+                })
+              });
+              //email the parent email address
+              function emailParents (cb) {
```

Comment:
This is faulty on multiple plans
First, it basically check if you don't have the same parent twice, because parentsEmailed will be resetted at every call.
Secondly, even if it was global and was growing, it would avoid a parent from receiving that his second child also has been booked
If yo uwant a "working" example, have a look at https://github.com/CoderDojo/cp-dojos-service/commit/91b32a2eace1d01b75a04451c60bfc51d67701ef#diff-cd7d1eaad452dcfd33578fdccdf76999R97 and how the data is built before